server/modules/category/service/categoryImportCSVParser.js

```diff
@@ -0,0 +1,151 @@
+const R = require('ramda')
+
+const CategoryLevel = require('../../../../common/survey/categoryLevel')
+const CategoryImportSummary = require('../../../../common/survey/categoryImportSummary')
+const StringUtils = require('../../../../common/stringUtils')
+
+const CSVReader = require('../../../utils/file/csvReader')
+
+const columnSuffixes = {
+  code: '_code',
+  label: '_label',
+  description: '_description',
+}
+
+const columnRegExpLabel = new RegExp(`^.*${columnSuffixes.label}(_[a-z]{2})?$`)
+const columnRegExpDescription = new RegExp(`^.*${columnSuffixes.description}(_[a-z]{2})?$`)
+
+const _readHeaders = filePath => new Promise((resolve, reject) => {
+  try {
+    const reader = CSVReader.createReader(
+      filePath,
+      headers => {
+        reader.cancel()
+        resolve(headers)
+      }
+    )
+
+    reader.start()
+  } catch (error) {
+    reject(error)
+  }
+})
+
+const createImportSummary = async (filePath) => {
+  //clean category levels
+  const headers = await _readHeaders(filePath)
+
+  const levelNames = headers.reduce(
+    (accLevelNames, header) => {
+      if (header.endsWith(columnSuffixes.code)) {
+        accLevelNames.push(header.substr(0, header.length - columnSuffixes.code.length))
+      }
+      return accLevelNames
+    },
+    []
+  )
+
+  const headersInfo = headers.reduce(
+    (acc, header) => {
+      if (header.endsWith(columnSuffixes.code)) {
+        const levelIndex = R.pipe(
```

Comment:
if you generate an object (see above comment) you don't need to iterate again on the headers to get it